source/common/stats/thread_local_store.h

```diff
@@ -16,6 +16,145 @@
 namespace Envoy {
 namespace Stats {
 
+/**
+ * Log Linear Histogram implementation per thread.
+ */
+class ThreadLocalHistogramImpl : public Histogram, public MetricImpl {
+public:
+  ThreadLocalHistogramImpl(const std::string& name, Store& parent, std::string&& tag_extracted_name,
+                           std::vector<Tag>&& tags)
+      : MetricImpl(name, std::move(tag_extracted_name), std::move(tags)), parent_(parent),
+        current_active_(0), flags_(0) {
+    histograms_[0] = hist_alloc();
+    histograms_[1] = hist_alloc();
+  }
+
+  ~ThreadLocalHistogramImpl() {
+    hist_free(histograms_[0]);
+    hist_free(histograms_[1]);
+  }
+  // Stats::Histogram
+  void recordValue(uint64_t value) override {
```

Comment:
suggest grouping together all the overrides and only inlining them if they fit on one line.  Then you can omit the inter-line spacing for the overrides.

beginMerge() below is not an override.

Reply:
sorry i misread the comment. I thought you were just asking to move all overrides to one place and did not pay attention to "inline" word - will take care of it in my next commit.